modules/4337/contracts/Safe4337Module.sol

```diff
@@ -211,6 +211,43 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
         operationHash = keccak256(operationData);
     }
 
+    /**
+     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not check the integrity of the signature encoding.
+     * @param signatures signatures data
+     * @param threshold Indicates the number of iterations to perform in the loop.
+     * @return result True if length check passes, false otherwise.
+     */
+    function _checkSignatureLength(bytes calldata signatures, uint256 threshold) internal pure returns (bool) {
+        uint256 offset = threshold * 0x41;
+
+        if (signatures.length < offset) return false;
+        bool pointsAtEnd = true;
+        for (uint256 i = 0; i < threshold; i++) {
+            /* solhint-disable no-inline-assembly */
+            /// @solidity memory-safe-assembly
+            assembly {
+                let signaturePos := mul(0x41, i)
+                // read the Safe signature type (uint8) from calldata.
+                let signatureType := byte(0, calldataload(add(signatures.offset, add(signaturePos, 0x40))))
+
+                // signatureType = 0 indicates that signature is a smart contract signature in Safe Signature Encoding
+                if iszero(signatureType) {
+                    // For Safe smart contract signature the s value points to the start of the signature data in the signatures
+                    // s value is relative to the signature data.
```

Comment:
nit: Comment is outdated as there is no "s" value anymore. Maybe you could say the "second word"?

Reply:
https://github.com/safe-global/safe-modules/pull/453/commits/979b6a4574003e6c1a0beaad4aefe9e8aa653c83